sdk/core/Azure.Core/tests/ClientDiagnosticsTests.Net50.cs

```diff
@@ -468,6 +470,38 @@ public void FailedStopsActivityAndWritesExceptionEventActivitySource()
             CollectionAssert.Contains(activity.Tags, new KeyValuePair<string, string>("Attribute2", "2"));
             CollectionAssert.Contains(activity.Tags, new KeyValuePair<string, string>("az.namespace", "Microsoft.Azure.Core.Cool.Tests"));
         }
+
+        [Test]
+        [NonParallelizable]
+        public void OpenTelemetryCompatibilityWithParentBasedSampling()
```

Comment:
please add a deterministic test instead.

I.e use 
- `AlwaysOff` sampler and check that no activities were created.

OR 

- use parent-based sampler and manually set `00-valid_trace_id-valid_span_id-00` as parentId and expect it to be sampled out. 
 
?